src/marqo/_httprequests.py

```diff
@@ -52,64 +64,106 @@ def send_request(
         with warnings.catch_warnings():
             if not self.config.cluster_is_remote:
                 warnings.simplefilter('ignore', InsecureRequestWarning)
-            try:
-                request_path = self.config.url + '/' + path
-                if isinstance(body, (bytes, str)):
-                    response = http_method(
-                        request_path,
-                        timeout=self.config.timeout,
-                        headers=req_headers,
-                        data=body,
-                        verify=to_verify
-                    )
-                else:
-                    response = http_method(
-                        request_path,
-                        timeout=self.config.timeout,
-                        headers=req_headers,
-                        data=json.dumps(body) if body else None,
-                        verify=to_verify
-                    )
-                return self.__validate(response)
-            except requests.exceptions.Timeout as err:
-                raise BackendTimeoutError(str(err)) from err
-            except requests.exceptions.ConnectionError as err:
-                raise BackendCommunicationError(str(err)) from err
+            
+            for attempt in range(max_retry_attempts + 1):
+                try:
+                    request_path = self.config.url + '/' + path
+                    if isinstance(body, (bytes, str)):
+                        response = http_method(
+                            request_path,
+                            timeout=self.config.timeout,
+                            headers=req_headers,
+                            data=body,
+                            verify=to_verify
+                        )
+                    else:
+                        response = http_method(
+                            request_path,
+                            timeout=self.config.timeout,
+                            headers=req_headers,
+                            data=json.dumps(body) if body else None,
+                            verify=to_verify
+                        )
+                    return self.__validate(response)
+                except requests.exceptions.Timeout as err:
+                    raise BackendTimeoutError(str(err)) from err
+                except requests.exceptions.ConnectionError as err:
+                    if (attempt == max_retry_attempts):
```

Comment:
Add unit tests for `send_request` with varying `max_retry_attempts` and `max_retry_backoff_seconds`